tests/docker_cluster.py

```diff
@@ -60,8 +66,34 @@ def __init__(self, master_host, slave_hosts,
     def all_hosts(self):
         return self.slaves + [self.master]
 
+    def all_internal_hosts(self):
+        """The difference between this method and all_hosts() is that
+        all_hosts() returns the unique, "outside facing" hostnames that
+        docker uses. On the other hand all_internal_hosts() returns the
+        more human readable host aliases for the containers used internally
+        between containers. For example the unique master host will
+        look something like 'master-07d1774e-72d7-45da-bf84-081cfaa5da9a',
+        whereas the internal master host will be 'master'.
+
+        Returns:
+            List of all internal hosts with the random suffix stripped out.
+        """
+        return [host.split('-')[0] for host in self.all_hosts()]
+
     def get_local_mount_dir(self, host):
-        return os.path.join(self.local_mount_dir, host)
+        return os.path.join(self.local_mount_dir,
+                            self.__get_unique_host(host))
+
+    def __get_unique_host(self, host):
+        matches = [unique_host for unique_host in self.all_hosts()
+                   if unique_host.startswith(host)]
+        if matches:
```

Comment:
So this method is ensuring that get_local_mount_dir will work on either master or master-124ab2312328?